examples/se_2_3_localization.cpp

```diff
@@ -0,0 +1,452 @@
+
+
+    // START TEMPORAL LOOP
+    //
+    //
+
+    // Make 10 steps. Measure up to three landmarks each time.
+    for (double t = 0; t < 10; t+=dt)
+    {
+        //// I. Simulation ###############################################################################
+
+        /// get current simulated state and measurements from previous step
+        auto R_k = X_simulation.rotation();
+        auto v_k = X_simulation.linearVelocity();
+        auto acc_k = alpha_prev + R_k.transpose()*g;
+
+        /// input vector
+        u_nom << (dt*(R_k.transpose())*v_k + 0.5*dt*dt*acc_k),  dt*omega_prev, dt*acc_k;
+
+        /// simulate noise
+        u_noise = u_sigmas * Array9d::Random();             // control noise
+        u_noisy = u_nom + u_noise;                          // noisy control
+
+        u_simu   = u_nom;
+        u_unfilt = u_noisy;
+
+        /// first we move - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+        X_simulation = X_simulation + u_simu;               // overloaded X.rplus(u) = X * exp(u)
+        /// update expected IMU measurements
+        alpha = alpha_const - X_simulation.rotation().transpose()*g; // update expected IMU measurement after moving
+
+        /// then we measure all landmarks - - - - - - - - - - - - - - - - - - - -
+        for (int i = 0; i < landmarks.size(); i++)
+        {
+            b = landmarks[i];                               // lmk coordinates in world frame
+
+            /// simulate noise
+            y_noise = y_sigmas * Array3d::Random();         // measurement noise
+
+            y = X_simulation.inverse().act(b);              // landmark measurement, before adding noise
+            y = y + y_noise;                                // landmark measurement, noisy
+            measurements[i] << y;                     // store for the estimator just below
+        }
+
+
+
+
+        //// II. Estimation ###############################################################################
+
+        /// get current state estimate to build the state-dependent control vector
+        auto R_k_est = X.rotation();
+        auto v_k_est = X.linearVelocity();
+        auto acc_k_est = alpha_prev + R_k_est.transpose()*g;
+
+        Eigen::Vector3d accLin = dt*(R_k_est.transpose())*v_k_est + 0.5*dt*dt*acc_k_est;
+        Eigen::Vector3d gLin = R_k_est.transpose()*g*dt;
+        Eigen::Matrix3d accLinCross = manif::skew(accLin);
+        Eigen::Matrix3d gCross = manif::skew(gLin);
+
+        u_est << accLin,  dt*omega_prev, dt*acc_k_est;
+        u_est += u_noise;
+
+        /// First we move - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+        X = X.plus(u_est, J_x, J_u);                        // X * exp(u), with Jacobians
+
+        // Prepare Jacobian
+        slantFk.setZero();
+        slantFk.block<3, 3>(0, 3) = accLinCross;
+        slantFk.block<3, 3>(0, 6) = Eigen::Matrix3d::Identity()*dt;
+        slantFk.block<3, 3>(6, 3) = gCross;
+        Fk = J_x + J_u*slantFk;
+
```

Comment:
Rename :
- `Fk`: leave as is if you want
- `J_x` --> `J_x_x` which is df/dx
- `J_u` --> `J_x_u`
- `slantFk` --> `J_u_x`
This results in L378 being:
- `Fk = J_x_x + J_x_u*J_u_x` which used the chain rule

Reply:
Thank you, @joansola,  for the suggestion! These names are rather quite informative than the former ones.
I have fixed this in https://github.com/artivis/manif/pull/154/commits/b00e27afe04fdf84e7fb052a6d796614d8da83b4.

Reply:
Yea that's the point! More information, easy to understand --> less errors!